server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -122,6 +129,14 @@ public Version indexVersionCreated() {
                 return queryShardContextSupplier;
             }
 
+            public DateFormatter getDateFormatter() {
```

Comment:
Probably worth javadoc - this is the format used for dynamic date detection and the default date format for date fields that don't have one set. Right?